### PR TITLE
Closes #14234 and updates text

### PR DESCRIPTION
**Pull request by moxiegirl**
- Carry of #14234
- Fixing excessive notes in text

Signed-off-by: Mary Anthony mary@docker.com

### CODE REVIEW

File: docs/userguide/dockervolumes.md [hunk 1]
```diff
@@ -95,39 +95,30 @@ if the volume is read/write.
...
-> **Note:** 
-> This is not available from a `Dockerfile` due to the portability
-> and sharing purpose of built images. The host directory is, by its nature,
-> host-dependent, so a host directory specified in a `Dockerfile` probably
-> wouldn't work on all hosts.
+If you are using Docker Machine on Mac or Windows, your Docker daemon has only limited access to your OS X or Windows filesystem. Docker Machine tries
+to auto-share your `/Users` (OS X) or `C:\Users` (Windows) directory.  So,
+you can mount files or directories on OS X using.
+
+```docker run -v /Users/<path:/<container path ...
```
Comment (thaJeztah) [inline docs/userguide/dockervolumes.md]: docker should start on a new line here, not directly after the "fence"

File: docs/userguide/dockervolumes.md [hunk 2]
```diff
@@ -95,39 +95,30 @@ if the volume is read/write.
...
+```
+
+On Windows, mount directories using:
+
+```docker run -v /c/Users/<path:/<container path ...`
```
Comment (thaJeztah) [inline docs/userguide/dockervolumes.md]: Same here

File: docs/userguide/dockervolumes.md [hunk 3]
```diff
@@ -95,39 +95,30 @@ if the volume is read/write.
...
+
+```docker run -v /Users/<path:/<container path ...
+```
+
+On Windows, mount directories using:
+
+```docker run -v /c/Users/<path:/<container path ...` 
+```
+
+All other paths come from your virtual machine's filesystem.  For example, if you are using Virtualbox some other folder available for sharing, you need to do additional work. In the case of VirtualBox you need to make the host folder available as a shared folder in Virtualbox. Then, you can mount it using the Docker `-v` flag.
```
Comment (thaJeztah) [inline docs/userguide/dockervolumes.md]: Nit; s/Virtualbox/VirtualBox/

Comment (thaJeztah) [inline docs/userguide/dockervolumes.md]: Also, could you wrap this line to be consistent with the rest of the document?

File: docs/userguide/dockervolumes.md [hunk 4]
```diff
@@ -95,39 +95,30 @@ if the volume is read/write.
...
-> way the source directory name is parsed. You can fix it by using a double slash at
-> the beginning of the source directory name as explained in [issue #12751](https://github.com/docker/docker/issues/12751)
-
-This is very useful for testing, for example we can
-mount our source code inside the container and see our application at work as
-we change the source code. The directory on the host must be specified as an
-absolute path and if the directory doesn't exist Docker will automatically
-create it for you.
+This commands mounts the host directory, `/src/webapp`, into the container at
+`/opt/webapp`.  If the path `/opt/webapp` already exists inside the container's image, its contents are replaced by the contents of `/src/webapp` on the host to stay consistent with the expected behavior of  the `mount` command.
```
Comment (thaJeztah) [inline docs/userguide/dockervolumes.md]: Could you wrap this line as well? <3

**Comment by thaJeztah** [comment]
Just some nits, LGTM otherwise, thanks Mary!

**Comment by SvenDowideit** [comment]
I wonder if we can modify the `its contents are replaced by the contents of` - some users have taken this to mean that the container image will be modified, when in actual fact, the bind-mounted directory hides the image directory - and when the user `docker commits` the container, that image will not contain the bind-mounted files, but rather the underlying original files.

**Comment by moxiegirl** [comment]
@thaJeztah for the merge.  Author LGTM

**Comment by thaJeztah** [comment]
LGTM, thanks @moxiegirl (and thanks, Sven; good point)

**Pull request merged**